src/utils/common/apiUtil.ts

```diff
@@ -50,23 +67,20 @@ export const getAlbumsURL = () => {
 // getFamilyPortalURL returns the endpoint for the family dashboard which can be used to
 // create or manage family.
 export const getFamilyPortalURL = () => {
-    if (process.env.NEXT_PUBLIC_ENTE_FAMILY_PORTAL_ENDPOINT !== undefined) {
-        return process.env.NEXT_PUBLIC_ENTE_FAMILY_PORTAL_ENDPOINT;
+    const familyURL = process.env.NEXT_PUBLIC_ENTE_FAMILY_PORTAL_ENDPOINT;
+    if (isDevDeployment() && familyURL) {
+        return familyURL;
     }
     return `https://family.ente.io`;
 };
 
-export const getUploadEndpoint = () => {
-    if (process.env.NEXT_PUBLIC_ENTE_UPLOAD_ENDPOINT !== undefined) {
-        return process.env.NEXT_PUBLIC_ENTE_UPLOAD_ENDPOINT;
-    }
-    return `https://uploader.ente.io`;
-};
-
 export const getSentryTunnelURL = () => {
     return `https://sentry-reporter.ente.io`;
 };
 
 const isDevDeployment = () =>
-    process.env.NEXT_PUBLIC_ENTE_WEB_ENDPOINT === window.location.origin ||
-    process.env.NEXT_PUBLIC_ENTE_ALBUM_ENDPOINT === window.location.origin;
+    runningInBrowser() &&
+    (process.env.NEXT_PUBLIC_ENTE_WEB_ENDPOINT === window.location.origin ||
+        process.env.NEXT_PUBLIC_ENTE_ALBUM_ENDPOINT ===
+            window.location.origin ||
+        process.env.NODE_ENV === 'development');
```

Comment:
should this be an and condition for process.env.NODE_ENV === 'development')

Reply:
If not, why is that? May be we can add some documentation for this method

Reply:
Yes, it is as it should be.

So the condition says that it's a dev deployment (and should use the environment override for ENDPOINTS ) in three cases 
when 
1. when the app opened is the staging web app, or
2. when the app opened is the staging album app, or
3. if the app is running locally (hence node_env is `development`)